Library/Homebrew/rubocops/cask/constants/stanza.rb

```diff
@@ -8,8 +8,7 @@ module Constants
       STANZA_GROUPS = [
         [:version, :sha256],
         [:language],
-        [:url, :appcast, :name, :desc, :homepage],
-        [:livecheck],
+        [:url, :appcast, :livecheck, :name, :desc, :homepage],
```

Comment:
I'm not sure where it should go. It's basically a replacement for `appcast`, so I thought it should be in the same place.

Another option would be a new group with `[:auto_updates, :livecheck]`.

Reply:
> My intention behind the previous grouping was to produce something like

That looks better to me. Having a block in the middle of lines (`appcast` position) is hard to follow. I’d even go so far as saying it should be right at the bottom, below everything else, because it’s a block that should remain mostly static (all going well).

Reply:
I made [the location/grouping for `livecheck` blocks in casks](https://github.com/Homebrew/brew/blob/b0a3b60292b7919f22a43d40e786ef35562d22df/Library/Homebrew/rubocops/cask/constants/stanza.rb#L12) as close to [the location in formulae](https://github.com/Homebrew/brew/blob/b0a3b60292b7919f22a43d40e786ef35562d22df/Library/Homebrew/rubocops/components_order.rb#L27) as possible and it would be nice to maintain consistency between core/cask. Keeping the `livecheck` block by itself also helps readability.

---

> Another option would be a new group with `[:auto_updates, :livecheck]`.

Currently `livecheck` is in its own group that comes before `auto_updates` and I think that makes sense.

It's not uncommon for us to have comments before the `livecheck` block (explaining an unusual situation). Placing `auto_updates` before `livecheck` in the same group would look weird when there are explanatory comments:

```ruby
auto_updates true
# Aften has moved from a version scheme like 0.07 to 0.0.8. We restrict
# matching to versions with three parts, since a version like 0.07 is parsed
# as 0.7 and seen as newer than 0.0.8.
livecheck do
  url :stable
  regex(%r{url=.*?/aften[._-]v?(\d+(?:\.\d+){2,})\.t}i)
end
```

Placing `auto_updates` after `livecheck` in the same group could lead to `auto_updates` being overlooked. I think the current setup is the most clear and having the cask location in line with formulae helps to keep it predictable between core/cask.

```ruby
# Aften has moved from a version scheme like 0.07 to 0.0.8. We restrict
# matching to versions with three parts, since a version like 0.07 is parsed
# as 0.7 and seen as newer than 0.0.8.
livecheck do
  url :stable
  regex(%r{url=.*?/aften[._-]v?(\d+(?:\.\d+){2,})\.t}i)
end
auto_updates true
```

Additionally, if you group `auto_updates` and `livecheck` together, `auto_updates` appears on its own line when there isn't a `livecheck` block. I think it's best to leave `auto_updates` at the top of the group that comes after `livecheck`. They're already near to one another and grouping them together seems like it would detract from readability rather than improve it.

---

> I’d even go so far as saying it should be right at the bottom, below everything else, because it’s a block that should remain mostly static (all going well).

Part of our thinking when originally selecting a location for the `livecheck` block in formulae was that it would appear near the URLs it uses (`homepage`, `url`/`stable`, `head` for formulae or `url`, `appcast`, `homepage` for casks). This is important because a `livecheck` block should be checked (and updated) when the URLs in a cask/formula notably changes (i.e., something more than a version bump).

Placing the `livecheck` block at the bottom of a cask would distance it from these URLs and make it more likely to be overlooked in this circumstance. Even with the `livecheck` block close to these URLs in homebrew/core formulae, we've already had problems with people changing the `stable` source and not updating the `livecheck` block, so it's better not to make this even easier to miss.

Reply:
> I made [the location/grouping for `livecheck` blocks in casks](https://github.com/Homebrew/brew/blob/b0a3b60292b7919f22a43d40e786ef35562d22df/Library/Homebrew/rubocops/cask/constants/stanza.rb#L12) as close to [the location in formulae](https://github.com/Homebrew/brew/blob/b0a3b60292b7919f22a43d40e786ef35562d22df/Library/Homebrew/rubocops/components_order.rb#L27) as possible and it would be nice to maintain consistency between core/cask.

Casks and formulae are fundamentally different in how they’re built, so I don’t think that matters and that it’s more important for them to be consistent and easy to parse within themselves.

> This is important because a `livecheck` block should be checked (and updated) when the URLs in a cask/formula notably changes (i.e., something more than a version bump).

But that convinces me.

> Keeping the `livecheck` block by itself also helps readability.

And that’s what I find most important anyway. I don’t think this should be grouped with any other stanza (including `auto_updates`). We don’t do it for any other block; they’re always kept separate.

Reply:
Reverted this now to the existing location.